tools/slang-unit-test/unit-test-math.cpp

```
@@ -0,0 +1,332 @@
+    SLANG_CHECK(Math::AreNearlyEqual(0.0, 0.0, 0.0));
+    SLANG_CHECK(Math::AreNearlyEqual(-7.25, -7.25, 1e-15));
+}
+
+SLANG_UNIT_TEST(mathAreNearlyEqualRelative)
+{
+    // Inputs within relative tolerance.
+    const double eps = 1e-9;
+    SLANG_CHECK(Math::AreNearlyEqual(1.0, 1.0 + 1e-12, eps));
+    SLANG_CHECK(!Math::AreNearlyEqual(1.0, 1.001, eps));
+
+    // Large magnitudes — relative tolerance applies.
```

Cppcheck (2.20.0)

[style] The function 'symmetricHash' is never used.

(unusedFunction)